MAIN PAGE

michalkrzem opened the pull request:
Products list to the home page moved

Commit:
MAIN PAGE

Products list to the home page moved

Commit:
MAIN PAGE HOTFIX - IMPORT UNUSED

ProductListView import but unused. I deleted it.

jacoor commented on Dshop/Dshop/urls.py:
```diff
@@ -34,6 +36,7 @@
     path(
         "sitemap.xml", sitemap, {"sitemaps": sitemaps}, name="django.contrib.sitemaps.views.sitemap",
     ),
+    path('', ProductListView.as_view(), name='products-list'),
```
I would do it differently. 
Remove changes in this file and check below.

jacoor commented on Dshop/Dshop/urls.py:
```diff
@@ -34,6 +36,7 @@
         "sitemap.xml", sitemap, {"sitemaps": sitemaps}, name="django.contrib.sitemaps.views.sitemap",
     ),
+    path('', ProductListView.as_view(), name='products-list'),
     path('products/', include('apps.products_catalogue.urls')),
```
```suggestion
    path('', include('apps.products_catalogue.urls')),
```

jacoor commented on Dshop/apps/products_catalogue/urls.py:
```diff
@@ -2,13 +2,12 @@
 
 
 urlpatterns = [
-    path('products_list/', ProductListView.as_view(), name='products-list'),
```
Bring this back, but remove "products_list" so it's 
```
path('', ProductListView.as_view(), name='products-list'),
```
This way, empty url (main page) points to ProductListView and all is set in one file.

jacoor commented on Dshop/apps/products_catalogue/urls.py:
```diff
@@ -2,13 +2,12 @@
 
 urlpatterns = [
-    path('products_list/', ProductListView.as_view(), name='products-list'),
     path('ceneo/', CeneoProductListView.as_view(), name='product-list'),
```
```suggestion
    path('products/ceneo/', CeneoProductListView.as_view(), name='product-list'),
```
Since all links except main page are prefixed with "products", let's also add it here.

jacoor reviewed (changes requested):
Good idea but it would make project hard to maintain. The changes I propose still keep produts urls in one place.

jacoor committed:
Merge branch 'dev' into feature/main_page

jacoor committed:
Merge branch 'dev' into feature/main_page

jacoor commented:
Hey
Any chance for update? Should be simple. 
And please, pull, I updated your branch from dev.

jacoor committed:
Merge branch 'dev' into feature/main_page

jacoor commented:
@michalkrzem Hey
Any chance for update? Should be simple.
And please, pull, I updated your branch from dev.

michalkrzem commented:
Sorry, i lost time 🥴

Commit:
MAIN PAGE SUGGESTED MODIFICATIONS

Remove 'products/' from Dshop/urls and set empty path included from
products_catalogue/urls.py

michalkrzem committed:
Merge branch 'dev' into feature/main_page

jacoor commented:
Merge when ready!
